include/exadg/incompressible_navier_stokes/postprocessor/line_plot_calculation.cpp

```diff
@@ -51,7 +51,32 @@ LinePlotCalculator<dim, Number>::setup(DoFHandler<dim> const & dof_handler_veloc
   data                 = line_plot_data_in;
 
   if(data.calculate)
+  {
     create_directories(data.line_data.directory, mpi_comm);
+
+    // Set up data to be requested on rank 0
+    if(Utilities::MPI::this_mpi_process(mpi_comm) == 0)
+    {
+      // collect all points to be evaluated
+      evaluation_points.clear();
+      for(typename std::vector<std::shared_ptr<Line<dim>>>::const_iterator line =
+            data.line_data.lines.begin();
+          line != data.line_data.lines.end();
+          ++line)
+      {
+        unsigned int n_points = (*line)->n_points;
+        // we consider straight lines with an equidistant distribution of points along the line
+        for(unsigned int i = 0; i < n_points; ++i)
+          evaluation_points.push_back((*line)->begin + double(i) / double(n_points - 1) *
+                                                         ((*line)->end - (*line)->begin));
+      }
+      evaluation_cache->reinit(evaluation_points,
+                               dof_handler_velocity->get_triangulation(),
+                               mapping_in);
+    }
+    else
+      evaluation_cache->reinit({}, dof_handler_pressure->get_triangulation(), mapping_in);
```

Comment:
I don't understand this. Why is `dof_handler_velocity` relevant in the `if`-branch and `dof_handler_pressure` in the `else` branch? If not, we should extract the triangulation above or even add to the interface of the class.

Very generally, could you explain the logic between the if and else branch (why is an else branch necessary and what is done there?, why is ``reinit()`` called with `{}` in the else branch?)?

Reply:
The logic is that MPI rank 0 requests the points to be evaluated (so it adds the points), whereas the other ranks to not request any point to be evaluated. Taking into account the other comment, is will be good to not call `reinit` in the `if/else` branches but collect the points to be evaluated.